### PR TITLE
Added details on working with single records from form submissions.

### PR DESCRIPTION
**Pull request by AndyButland**
This is a small update coming from an escalated support request I had, which was solved using an interface that we hadn't documented.  So wanted to make sure we have a public reference to it.

It can be published at any time.

### CODE REVIEW

File: Add-ons/UmbracoForms/Developer/Working-With-Data/index-v8.md [hunk 1]
```diff
@@ -6,10 +6,13 @@ meta.Description: "Developer documentation on working with Forms record data."
...
-From Umbraco Forms `v8.2` includes some helper methods that return records of a given form, which can be used to output records in your templates using razor.
+Umbraco Forms includes some helper methods that return records of a given form, which can be used to output records in your templates using Razor.
 
-## Available methods
-The methods can be found by injecting the `Umbraco.Forms.Core.Services.IRecordReaderService` interface. For performance reasons, all these methods are paged.
+## Working with records in bulk
+
+The methods can be found by injecting the `Umbraco.Forms.Core.Services.IRecordReaderService` interface.
+
+For performance reasons, all these methods are paged and retrieve their values from an underying Examine search index.
```
Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Working-With-Data/index-v8.md:15]: underlying

File: Add-ons/UmbracoForms/Developer/Working-With-Data/index-v8.md [hunk 2]
```diff
@@ -79,11 +82,12 @@ Dictionary<Guid, RecordField> RecordFields
 ```
 
 In order to access custom form fields, these are available in the `RecordFields` property.
+
 Furthermore there exists an extension method named `ValueAsString` on  `IRecord` in `Umbraco.Forms.Core.Services`, such that you can get the value as string given the alias of the field.
```
Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Working-With-Data/index-v8.md:86]: Furthermore, 

as a string

File: Add-ons/UmbracoForms/Developer/Working-With-Data/index-v8.md [hunk 3]
```diff
@@ -97,19 +101,41 @@ Sample script that is outputting comments using a form created with the default
...
+            <span>said</span>
+            <p>@record.ValueAsString("comment")</p>
+        </li>
     }
 </ul>
 ```
+
+## Working with single records
+
+If you have a use-case where you want to be able to retrieve the single record created from a form submission by a user, then a different interface, that provides access via database requests, should be used.  The interface is `IRecordStorage`, found in the `Umbraco.Forms.Core.Data.Storage` namespace.
```
Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Working-With-Data/index-v8.md:127]: use case

File: Add-ons/UmbracoForms/Developer/Working-With-Data/index-v8.md [hunk 4]
```diff
@@ -97,19 +101,41 @@ Sample script that is outputting comments using a form created with the default
...
+
+By using this interface you avoid any timing issues caused by the Examine indexes used in the previous examples not yet being populated.
+
+### GetRecordByUniqueId
+
+```csharp
+Record GetRecordByUniqueId(Guid uniqueId, Form form);
+```
+
+Returns a single record based on it's unique Id and form.
```
Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Working-With-Data/index-v8.md:137]: its

**Review by eshanrnh — COMMENTED** [review]
Looks good, @AndyButland. Just a couple of comments 🙂

File: Add-ons/UmbracoForms/Developer/Working-With-Data/index-v8.md (hunk 1 above)
Comment (AndyButland) [reply Add-ons/UmbracoForms/Developer/Working-With-Data/index-v8.md]: Thanks - have updated this and the other corrections.

**Review by eshanrnh — APPROVED** [review]
Merging it in 🙂

**Pull request merged**